File: server/guestregistration/src/main/java/com/exadel/guestregistration/services/EmployeeService.java
package com.exadel.guestregistration.services;

import com.exadel.guestregistration.exceptions.EmployeeException;
import com.exadel.guestregistration.exceptions.NotFoundException;
import com.exadel.guestregistration.models.Employee;

import java.util.List;

public interface EmployeeService {
    List<Employee> findAllEmployees();
    List<Employee> findAllInSpecifOffice(String id);
    Employee create(Employee employee) throws EmployeeException;
    Employee update(Employee employee) throws NotFoundException, EmployeeException;
    void delete(String id) throws NotFoundException;
    Employee findEmployeeById(String id) throws NotFoundException;
    List<Employee> findEmployees(String officeId, String text);
    boolean checkEmployee(String agentId);
    List<Employee>findEmployeesOfFullList(String text);
    List<Employee>findEmployeesByCardId(String cardId);   
}

File: client/guestregistration/src/app/services/employee.service.ts
import { Injectable } from '@angular/core';
import { Employee } from '../classes/employee';
import {Observable} from 'rxjs';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {BehaviorSubject} from 'rxjs';

const httpOptions = {
  headers: new HttpHeaders({'Content-Type':'application/json'}),
  withCredentials: true
};

@Injectable()
export class EmployeeService {
  private baseUrl = 'http://localhost:8080';

  private messageSource = new BehaviorSubject<string>("default message");
  currentMessage = this.messageSource.asObservable()

  constructor(private http: HttpClient) {}

  changeMessage(editing_employee_id: string) {
    this.messageSource.next(editing_employee_id);
  }

  getAllEmployees() {
    return this.http.get(this.baseUrl + '/api/employees/getAll', httpOptions);
  }

  getEmployees(id: string) {
    return this.http.get(this.baseUrl + '/api/employees/getAll/' + id, httpOptions);
  }

  getEmployee(id: string) {
    return this.http.get(this.baseUrl + '/api/employees/' + id, httpOptions);
  }

  createEmployee(employeeData: Object): Observable<Object> {
    return this.http.post(this.baseUrl + '/api/employees/create', employeeData, httpOptions);
  }

  updateEmployee(employeeData: Employee) {
    return this.http.put(this.baseUrl + '/api/employees/update/' + employeeData.id, employeeData, httpOptions)
  }

  deleteEmployee(id: string): Observable<{}> {
    return this.http.delete(this.baseUrl + '/api/employees/delete/' + id, httpOptions);
  }

  findEmployees(officeId: string, text: string) {
    return this.http.get(this.baseUrl + '/api/employees/find/' + officeId + '/'  + text, httpOptions);
  }

  findEmployeeInFullList(text: string) {
    return this.http.get(this.baseUrl + '/api/employees/find/' + text, httpOptions);
  }

  checkEmployee(agentId: string) {
    return this.http.get(this.baseUrl + '/api/employees/check/' + agentId, httpOptions);
  }

}
